### ISSUE TITLE
Expose KernelConnection publicly

### ISSUE THREAD

**Issue by tkrabel-db** [jupyterlab/jupyterlab#12155]
### Affiliation Disclaimer

I work for Databricks, and my request is related to work we currently do there. 

### TLDR

We want to support ipywidgets on our platform while reusing as many open source components possible. Due to architectural differences (e.g. we run all third party FE code in iframes and we have more Databricks specific components between the Browser and the Python kernel), we cannot simply use the `KernelManager` as provided in the `examples/` folder. instead, we need to reuse the `KernelConnection` directly.

### Problem

* I would like the KernelConnection to be accessible from a public AMD module because I need to use this components inside Databricks.

### Proposed Solution

* Export `KernelConnection`

**Comment by tkrabel-db** [comment]
CC @jasongrout
